kroxylicious/src/main/java/io/kroxylicious/proxy/internal/admin/PrometheusMetricsEndpoint.java

```diff
@@ -22,10 +22,7 @@ public class PrometheusMetricsEndpoint implements Function<HttpRequest, HttpResp
     private final PrometheusMeterRegistry registry;
 
     public PrometheusMetricsEndpoint(MeterRegistries registries) {
-        if (registries.maybePrometheusMeterRegistry().isEmpty()) {
-            throw new IllegalStateException("Attempting to configure a prometheus endpoint but no Prometheus registry available");
-        }
-        this.registry = registries.maybePrometheusMeterRegistry().get();
+        this.registry = registries.maybePrometheusMeterRegistry().orElseThrow(() -> new IllegalStateException("Attempting to configure a prometheus endpoint but no Prometheus registry available"));
```

Comment:
This one was actually quite subtle. Sonar highlights the need to call isPresent on line 25 because it can't ensure that return value from `registries.maybePrometheusMeterRegistry()` returns the same optional on each invocation. 

In reality we do and we could ignore the warning but given the fix is so simple it seems better to pander to Sonar than depend on the current implementation of `maybePrometheusMeterRegistry`.